simtools/configuration/configurator.py

```diff
@@ -277,6 +277,10 @@ def _fill_from_environmental_variables(self):
                 env_variable_to_read = f"SIMTOOLS_{key.upper()}"
                 if value is None:
                     _env_dict[key] = os.environ.get(env_variable_to_read)
+                if key in _env_dict and _env_dict[key] is not None:
```

Comment:
Why a separate for loop? We don't read env vars anywhere else so I think it can be in the same loop.

Reply:
The reason is that `os.environ.get(env_variable_to_read)` can return None, so we need to check again before applying the strip command.

Reply:
Sure, but you can do 
```
for key, value in self.config.items():
    # environmental variables for simtools should always start with SIMTOOLS_
    env_variable_to_read = f"SIMTOOLS_{key.upper()}"
    if value is None:
        env_value = os.environ.get(env_variable_to_read)
        if env_value is not None:
            env_value = env_value.split("#")[0].strip().replace('"', "").replace("'", "")
        _env_dict[key] = env_value
```
This would save the extra for loop and keep a `None` value if that is what is provided. 
(I included in the example above also the fixes in the next comment.)

Reply:
OK,, I've changed it accordingly (although I don't really see the differences - same amount of if statements...)